cmd/archive-access-api/main.go

```diff
@@ -141,7 +145,21 @@ func run() int {
 		}
 		log.Info().Msg("Flow Access API Server stopped")
 	}()
+	go func() {
+		if !metricsEnabled {
+			return
+		}
 
+		log.Info().Msg("metrics server starting")
+		server := metrics.NewServer(log, flagMetricsAddr)
+		err := server.Start()
+		if err != nil {
+			log.Warn().Err(err).Msg("metrics server failed")
+			close(failed)
+		}
```

Comment:
Should you return inside the `if` statement here to avoid calling `close()` twice? Maybe that's not a problem but wanted to point out.